src/pages/iou/MoneyRequestCategoryPage.js

```diff
@@ -83,11 +84,26 @@ MoneyRequestCategoryPage.displayName = 'MoneyRequestCategoryPage';
 MoneyRequestCategoryPage.propTypes = propTypes;
 MoneyRequestCategoryPage.defaultProps = defaultProps;
 
-export default withOnyx({
-    report: {
-        key: ({route}) => `${ONYXKEYS.COLLECTION.REPORT}${lodashGet(route, 'params.reportID', '')}`,
-    },
-    iou: {
-        key: ONYXKEYS.IOU,
-    },
-})(MoneyRequestCategoryPage);
+export default compose(
+    withOnyx({
+        iou: {
+            key: ONYXKEYS.IOU,
+        },
+    }),
+    // This is a temporary hack to forward forward iou value to next getter.
+    // There is a problem that with one withOnyx it does not forward value from first to second.
+    // eslint-disable-next-line rulesdir/no-multiple-onyx-in-file
+    withOnyx({
+        report: {
+            key: ({route, iou}) => {
+                let reportID = lodashGet(route, 'params.reportID', '');
+                if (!reportID) {
```

Comment:
Wasn't this already fixed with https://github.com/Expensify/App/pull/27797?

Reply:
Not actually. This is a fix for split bills, not for money requests. I need to update a comment to clarify.

Reply:
Can't the same fix be applied?

Reply:
Good question. Let me try to play with it 🙂

Reply:
Actually, it was a good idea to unify fixes. I've decided to use a fallback `reportID` from **iou** `participants`.

Reply:
Done - 0bcda51.

Reply:
Looks good to me, it's funny because this was my first fix when I implemented this for the tags on the first PR. The only issue is that the report ID won't be on the URL, but since IOU is persisted, it might not cause issues.